### PR TITLE
Zs 43 refactor ad hoc management service use correlation id provider service impl

### PR DESCRIPTION
**Pull request by robinbakkeruscgi**

### CODE REVIEW

File: osgp-domain-core/src/main/java/com/alliander/osgp/domain/core/services/CorrelationIdProviderUUIService.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package com.alliander.osgp.domain.core.services;
+
+import java.util.UUID;
+
+import org.springframework.stereotype.Service;
+
+@Service
+public class CorrelationIdProviderUUIService {
```
Comment (robindenadel) [inline osgp-domain-core/src/main/java/com/alliander/osgp/domain/core/services/CorrelationIdProviderUUIService.java]: implement CorrelationIdProviderService

File: osgp-domain-core/src/main/java/com/alliander/osgp/domain/core/services/CorrelationIdProviderTimestampService.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+package com.alliander.osgp.domain.core.services;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import org.springframework.context.annotation.Primary;
+import org.springframework.stereotype.Service;
+
+@Service
+@Primary
```
Comment (robindenadel) [inline osgp-domain-core/src/main/java/com/alliander/osgp/domain/core/services/CorrelationIdProviderTimestampService.java]: I don't think this service should be the primary one.

File: osgp-adapter-domain-microgrids/src/main/java/com/alliander/osgp/adapter/domain/microgrids/application/services/AdHocManagementService.java [hunk 3]
```diff
@@ -47,6 +46,9 @@
     @Autowired
     private Integer lastCommunicationUpdateInterval;
 
+    @Autowired
+    private CorrelationIdProviderUUIService correlationIdProviderUUIService;
```
Comment (robindenadel) [inline osgp-adapter-domain-microgrids/src/main/java/com/alliander/osgp/adapter/domain/microgrids/application/services/AdHocManagementService.java]: If you want to work with the @Primary annotation, this field could be of the type CorrelationIdProviderService

File: osgp-domain-core/src/main/java/com/alliander/osgp/domain/core/services/CorrelationIdProviderUUIService.java (hunk 1 above)
Comment (robinbakkeruscgi) [inline osgp-domain-core/src/main/java/com/alliander/osgp/domain/core/services/CorrelationIdProviderUUIService.java]: done, note also renamed the file.

**Pull request merged**